branding_company/model/branding_company.py

```diff
@@ -30,6 +29,29 @@ class BrandingCompany(models.Model):
         supporting models and reports.
         """
 
+    def get_partner_branding_company(self, partner_id):
+        """Returns recordset with branding company for partner or False."""
+        if not partner_id:
+            return False
+        partner_model = self.env['res.partner']
+        partner_obj = partner_model.browse(partner_id)
+        return partner_obj and partner_obj.branding_company_id or False
```

Comment:
just in case my empty recordset ramblings are not clear: I mean `self.env.browse([])` - this behaves like false, but you can access fields on it and get false back, which saves you a lot of checks afterwards